bindings_test.go

```diff
@@ -32,3 +33,84 @@ func TestCreateDataChunk(t *testing.T) {
 	chunk := CreateDataChunk(types)
 	defer DestroyDataChunk(&chunk)
 }
+
+func TestArrow(t *testing.T) {
+	defer VerifyAllocationCounters()
+
+	var config Config
+	defer DestroyConfig(&config)
+	if CreateConfig(&config) == StateError {
+		t.Fail()
+	}
+
+	var db Database
+	defer Close(&db)
+	var errMsg string
+	if OpenExt(":memory:", &db, config, &errMsg) == StateError {
+		require.Empty(t, errMsg)
+	}
+
+	var conn Connection
+	defer Disconnect(&conn)
+	if Connect(db, &conn) == StateError {
+		t.Fail()
+	}
+
+	var res Result
+	defer DestroyResult(&res)
+	if Query(conn, `FROM (
+		VALUES (1, 'foo'), (2, 'bar'), (3, 'baz')
+	) AS t(a, b)
+	`, &res) == StateError {
+		t.Fail()
+	}
+
+	colCount := ColumnCount(&res)
+	require.Equal(t, 2, int(colCount))
+
+	names := make([]string, colCount)
+	types := make([]LogicalType, colCount)
+	for i := range colCount {
+		names[i] = ColumnName(&res, i)
+		types[i] = ColumnLogicalType(&res, i)
+	}
+	// get arrow options
+	opt := ArrowOptions{}
```

Comment:
I think we need to defer destroying these.

Reply:
done